core/src/main/scala/org/http4s/MessageFailure.scala

```diff
@@ -92,12 +92,17 @@ sealed abstract class UnsupportedMediaTypeFailure(expected: Set[MediaRange]) ext
 final case class MediaTypeMissing(expected: Set[MediaRange])
   extends UnsupportedMediaTypeFailure(expected)
 {
-  val message = s"Decoder is unable to decode a Message without a MediaType. Expected media ranges: $expected"
+  val expectedMsg = expected.map(_.renderString).mkString(", ")
+  val message = s"No media type specified in Content-Type header. Expected one of the following media ranges: $expectedMsg"
 }
 
 /** Indicates that no [[EntityDecoder]] matches the [[MediaType]] of the [[Message]] being decoded */
 final case class MediaTypeMismatch(messageType: MediaType, expected: Set[MediaRange])
   extends UnsupportedMediaTypeFailure(expected)
 {
-  def message = s"$messageType is not a supported media type. Decoder accepts one of the following media ranges: $expected"
+  def message = {
+    val actual = messageType.renderString
+    val expectedMsg = expected.map(_.renderString).mkString(", ")
+    s"$actual is not a supported media type. Expected one of the following media ranges: $expectedMsg"
```

Comment:
I think [this parsing rule](https://github.com/http4s/http4s/blob/master/core/src/main/scala/org/http4s/parser/Rfc2616BasicRules.scala#L54-L56) makes it pretty hard to craft a header that, if served back to the user, would be malicious javascript, but does "being pretty sure it would be probably fairly hard" qualify as safe?